EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/DependenciesPanel.java

```diff
@@ -303,24 +329,76 @@ private void addListOfDependencies() {
+                ArrayList<String> keys = new ArrayList<>(dependenciesInput.keySet());
+                dependencyValues = new ArrayList<>();
+                for (String key : keys) {
+                    dependencyValues.add(new DependencyInput(key));
+                }
+                return dependencyValues;
+            }
+        };
+        listContainer = new WebMarkupContainer("dependenciesContainer");
+        listContainer.setOutputMarkupId(true);
+        Model selectAllModel = new Model<Boolean>() {
+            @Override
+            public Boolean getObject() {
+                return panel.allSelected;
+            }
+            @Override
+            public void setObject(Boolean bool) {
+                panel.allSelected = bool;
             }
         };
+        listContainer.add(new AjaxCheckBox("selectAllCheckbox", selectAllModel) {
+            @Override
+            protected void onUpdate(AjaxRequestTarget target) {
+                for (DependencyInput input : dependencyValues) {
+                    input.setSelected(panel.allSelected);
+                }
+                listview.removeAll();
+                updateDependenciesMapForCurrentSelection();
+                target.add(listContainer);
+            }
+        });
         listview = new ListView<String>("dependenciesListview", list) {
             @Override
             protected void populateItem(ListItem item) {
-                final String key = (String) item.getModelObject();
-                item.add(new Label("baseAttributeValue", key));
-                item.add(new Label("dependentAttributeValue", dependenciesInput.get(key)));
+                final DependencyInput key = (DependencyInput) item.getModelObject();
+                item.add(new Label("baseAttributeValue", key.baseValue));
+                item.add(new Label("dependentAttributeValue", dependenciesInput.get(key.baseValue)));
+                IModel<Boolean> dependencyInputModel = new Model<Boolean>() {
+                    @Override
+                    public Boolean getObject() {
+                        if (panel.allSelected) { return true; }
+                        for (DependencyInput input : dependencyValues) {
+                            if (input.baseValue.equals(key.baseValue)) {
+                               return input.getSelected();
+                            }
+                        }
+                        return false;
+                    }
+                    @Override
+                    public void setObject(Boolean bool) {
+                        for (DependencyInput input : dependencyValues) {
+                            if (input.baseValue.equals(key.baseValue)) {
+                                input.setSelected(bool);
+                            }
+                        }
+                    }
+                };
+                item.add(new AjaxCheckBox("deleteCheckbox", dependencyInputModel) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This class has 8 parents which is greater than 5 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)